provider/devices/common.go

```diff
@@ -108,9 +112,28 @@ func setupDevices() {
 		dbDevice.LastUpdatedTimestamp = 0
 		dbDevice.IsResetting = false
 		dbDevice.InitialSetupDone = false
-		dbDevice.StreamTargetFPS = 15
-		dbDevice.StreamJpegQuality = 75
-		dbDevice.StreamScalingFactor = 50
+
+		// Get the DeviceStreamSettings for the current device
+		deviceStreamSettings, err := db.GetDeviceStreamSettings(dbDevice.UDID)
+		if err != nil {
+			logger.ProviderLogger.LogError("setupDevices", fmt.Sprintf("Failed to retrieve stream settings for device `%s`: %v", dbDevice.UDID, err))
+			continue
+		}
+
+		// Check if the DeviceStreamSettings exist
+		if deviceStreamSettings.UDID == "" {
```

Comment:
Addition to [this](https://github.com/shamanec/GADS/pull/145/files#r1933758086). If we return an error when the there is no document directly then we can skip this check and directly fallback and do `updateDeviceWithGlobalSettings` above

Reply:
I think this was already adjusted in the previous review ([commit](https://github.com/shamanec/GADS/pull/145/commits/1ac38527d12822811cf856f95c3f1a73be0d0367))